Feat listing entry jsx

gfncodes opened the pull request:
Close #54 
Close #55 
Fix #63 

Refactor listingEntry to display listingEntry on one line.

Commit:
Rename listing.jsx to listingEntry.jsx

- This rename is inline with agreed-upon structure of client react
  component layout.
- Also tweak the appearance of listingEntry component to display a
  listingEntry on three lines.

Commit:
Refactor index.jsx to render Listings component

- Currently index.jsx renders a Listing (a single listing).  Make it
  renders Listings (a component comprised of an array of Listing
  components).
- Also add mock data to index.jsx (an array of two sample listings).

Commit:
Implement Listings component (listings.jsx)

- Implement Listings, a React component that renders an array of
  individual ListingEntry components.

Commit:
Refactor ListingEntry to display entry on 1 line

- Currently a ListingEntry takes up three lines.  Refactor to make the
  ListingEntry take up one line, and arrange the listing entry fields as
  follows: date, price, location, title (do NOT show the desription
  anymore).

lensvelt commented:
Comments:
- update props.listing.date to props.listing.createdAt
- change CSS class name conventions to 'lowercase-hyphen'

Commit:
Refactor listingEntry CSS class names

- Change ListingEntry React component class names to snake-case.
- Also change ListingEntry "date" field to "createdAt" field
  ("createdAt" will be actual field from Listings database table).

lensvelt commented:
Closes #66 

Review pull request - all changes effected, looks great!
